### ISSUE TITLE
It seems that `error` doesn't handle error properly

### ISSUE THREAD

**Issue by timnew** [petkaantonov/bluebird#729]
I found some wield behavior when I introduced `.error` into my promise chain.

Here is the simplified example:

```coffeescript
it 'should fail but passed', ->
   Promise.try ->
     Promise.reject new Error()
  .error console.log
```
I expect this test to fail in `mocha`, but it passed with nothing printed to console.

But the test failed immediately after I removed the `.error` handler

```coffeescript
it 'should fail', ->
   Promise.try ->
     Promise.reject new Error()
```

Or it also fails if I throw the exception instead of calling `.reject`.

```coffeescript
it 'should fail', ->
   Promise.try ->
     throw new Error()
  .error console.log
```

So I assume it is some kind of bug.

**Comment by benjamingr** [comment]
`.error` only catches `OperationalError` and not all regular `Error`s, this is useful in order to avoid silencing programmer errors. It is recommended that all your business logic errors extend `OperationalError`s.

This is pretty clear in the first line of the documentation of `Error`.

**Comment by spion** [comment]
Just to add, using `Promise.reject` results with the error being modified to have the operational error flag, while `throw` doesn't do that.

This is legacy behaviour and will be fixed with 3.0
